### DISCUSSION TITLE
Timestamp date type with optional time

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#12252] [Feature Requests]
I want to propose a new feature.

A timestamp data type **with optional time**.

The use case is when you need to storage, for example, an event datetime.
At the beginning we only know the day of the event. So the time would have to be empty/null.

And later, when we know the time, we can update the field to add the time and have the complete timestamp.

Ideally it is always the same field, because if you divide it into two... The queries and the Directus app display options become more complicated.

**Comment by abdonrd** [comment]
Maybe we could have like an `Unknown time` button:

| <img width="421" alt="Screenshot 2022-03-20 at 12 49 06" src="https://user-images.githubusercontent.com/1007051/159160789-9eadd4f2-75f9-4d1a-8db6-ff6dd5eb139a.png"> | <img width="403" alt="Screenshot 2022-03-20 at 12 45 56" src="https://user-images.githubusercontent.com/1007051/159160701-f82b28c0-ec29-414e-98a1-7b0ca665198e.png"> |
|------|------|

**Comment by azrikahar** [comment]
Interesting use case indeed. Assuming it is an actual `timestamp` field on the database side, how can we discern between unknown and actual times? Granted I have yet to actually test this so I could be wrong here, but I reckon "unknown" time will still be saved as something like `YYYY-MM-DD 00:00:00` (just an example) right? But then 12:00 am should still be a valid time input by user 🤔

**Reply by abdonrd** [reply]
Exactly! I was thinking of something like that!

**Reply by abdonrd** [reply]
Maybe with something like `YYYY-MM-DD 00:00:01` to allow the 12:00.
Probably setting the seconds to `:01` is rare enough.

**Reply by azrikahar** [reply]
It could be rare, but I think that may yield some odd behaviors with use cases like telemetry or IoT where timestamps are rather sensitive. Plus it may also interfere with date searches or even aggregations (and insights) in general. Tough one!

Maybe an alternative is to allow toggling between date/datetime/time picker independent of the actual field type, which you can then change the picker via field conditions based on a checkbox or status dropdown like booked vs confirmed etc.

**Reply by abdonrd** [reply]
> It could be rare, but I think that may yield some odd behaviors with use cases like telemetry or IoT where timestamps are rather sensitive. Plus it may also interfere with date searches or even aggregations (and insights) in general. Tough one!

Well... For use cases where you need precision, I guess you shouldn't use this "timestamp with optional time" field.

But yeah, I don't know what would be the best solution.

> Maybe an alternative is to allow toggling between date/datetime/time picker independent of the actual field type, which you can then change the picker via field conditions based on a checkbox or status dropdown like booked vs confirmed etc.

This way you would end up having two fields, right?
The date with time and the date without time.
I think that would complicate the queries and the app display options.

**Reply by benhaynes** [reply]
I agree with @azrikahar. While it might be slightly more work to use two fields, that is just how SQL works, and it doesn't make sense for us to choose a random time that "doesn't count" (even if flagged in an option). However, the best part of Directus is how you can always create your own interface extension for these edge-cases... so I'd recommend going that route.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
